src/graph/UpdateEdgeExecutor.cpp

```diff
@@ -190,68 +190,36 @@ void UpdateEdgeExecutor::finishExecution(storage::cpp2::UpdateResponse &&rpcResp
         rows.back().set_columns(std::move(row));
     }
     resp_->set_rows(std::move(rows));
-    doFinish(Executor::ProcessControl::kNext);
 }
 
-
 void UpdateEdgeExecutor::setupResponse(cpp2::ExecutionResponse &resp) {
     if (resp_ == nullptr) {
         resp_ = std::make_unique<cpp2::ExecutionResponse>();
     }
     resp = std::move(*resp_);
 }
 
-
-void UpdateEdgeExecutor::insertReverselyEdge(storage::cpp2::UpdateResponse &&rpcResp) {
-    std::vector<storage::cpp2::Edge> edges;
-    storage::cpp2::Edge reverselyEdge;
-    reverselyEdge.key.set_src(edge_.dst);
-    reverselyEdge.key.set_dst(edge_.src);
-    reverselyEdge.key.set_ranking(edge_.ranking);
-    reverselyEdge.key.set_edge_type(-edge_.edge_type);
-    // TODO(heng) For upsert, we should take the properties
-    reverselyEdge.props = "";
-    edges.emplace_back(reverselyEdge);
-    auto future = ectx()->getStorageClient()->addEdges(spaceId_, std::move(edges), false);
-    auto *runner = ectx()->rctx()->runner();
-    auto cb = [this, updateResp = std::move(rpcResp)] (auto &&resp) mutable {
-        auto completeness = resp.completeness();
-        if (completeness != 100) {
-            // Very bad, it should delete the upsert positive edge!!!
-            doError(Status::Error(
-                        "Insert the reversely edge(%s) `%ld->%ld@%ld' failed when update.",
-                        edgeTypeName_->c_str(),
-                        edge_.dst, edge_.src, edge_.ranking));
-            return;
-        }
-        this->finishExecution(std::move(updateResp));
-    };
-    auto error = [this] (auto &&e) {
-        auto msg = folly::stringPrintf(
-                "Insert reversely edge(%s) `%ld->%ld@%ld' exception when update: %s",
-                edgeTypeName_->c_str(),
-                edge_.dst, edge_.src, edge_.ranking,
-                e.what().c_str());
-        LOG(ERROR) << msg;
-        // Very bad, it should delete the upsert positive edge!!!
-        doError(Status::Error(std::move(msg)));
-    };
-    std::move(future).via(runner).thenValue(cb).thenError(error);
-}
-
-
-void UpdateEdgeExecutor::execute() {
-    FLOG_INFO("Executing UpdateEdge: %s", sentence_->toString().c_str());
-    std::string filterStr = filter_ ? Expression::encode(filter_) : "";
-    auto returns = getReturnColumns();
+void UpdateEdgeExecutor::updateEdge(bool reversely) {
+    std::string filterStr = "";
+    std::vector<std::string> returns;
+    if (reversely) {
+        auto src = edge_.src;
+        auto dst = edge_.dst;
+        edge_.set_src(dst);
+        edge_.set_dst(src);
+        edge_.set_edge_type(-edge_.edge_type);
+    } else {
+        filterStr = filter_ ? Expression::encode(filter_) : "";
```

Comment:
So the filter just processing when the edge is not reverse ?

Reply:
When updating the reverse edges, it means the out edges have been updated succeeded. So we don't need to check the filter again.